py/mattak/backends/uproot/dataset.py

```diff
@@ -215,33 +215,28 @@ def wfs(self, calibrated : bool =False) -> numpy.ndarray:
     def _iterate(self, start, stop, calibrated, max_in_mem,
                  selector: Optional[Callable[[mattak.Dataset.EventInfo], bool]] = None) -> Tuple[mattak.Dataset.EventInfo, numpy.ndarray]:
 
-        current_start = -1 
-        current_stop = -1
-        w = None
-        e = None
-        i = start 
-        preserve_entries = self.entry
-        if self.multiple:
-            preserve_entries = (self.first, self.last) 
-        while i < stop: 
-            if i >= current_stop:
-                current_start = i
-                current_stop = min(i+max_in_mem, stop)
-                self.setEntries((current_start, current_stop)) 
-                w = self.wfs(calibrated)
-                e = self.eventInfo()
-                self.setEntries(preserve_entries) # hide that we're modifying these 
-
-            rel_i = i - current_start
-            i += 1
-            if selector is not None:
-                if selector(e[rel_i]):
-                    yield e[rel_i], w[rel_i]
-            else:
-                yield e[rel_i], w[rel_i]
+        # determine in how many batches we want to access the data given how much events we want to load into the RAM at once
+        n_batches = (stop - start) // max_in_mem + 1
```

Comment:
does this spuriously add 1 if stop - start is a multiple of max_in_mem?

Reply:
Oh, it does indeed. I experienced this in my test (realising that now) but it did not cause a problem

Reply:
I use math.ceil now which should solve the problem.